core/deployment/src/main/java/io/quarkus/deployment/ExtensionLoader.java

```diff
@@ -963,6 +967,15 @@ public String toString() {
         return chainConfig;
     }
 
+    protected static List<Method> getMethods(Class<?> clazz) {
+        List<Method> declaredMethods = new ArrayList<>();
+        if (!clazz.getName().equals(Object.class.getName())) {
+            declaredMethods.addAll(getMethods(clazz.getSuperclass()));
```

Comment:
I generally *don't*, obviously, but i'm happy to break that up if you want.  It doesn't seem to affect anyone else as it would only affect someone who'd defined `BuildStep` on a class with a parent which doesn't seem to be happening anywhere else.  All the other tests I've run still pass, though.